deployer/commands/exec/infra_components.py

```diff
@@ -112,27 +121,33 @@ def root_homes(
         },
     }
 
-    cmd = [
+    # Command to exec into pod
+    exec_cmd = [
         "kubectl",
         "-n",
         hub_name,
-        "run",
-        "--rm",  # Remove pod when we're done
-        "-it",  # Give us a shell!
-        "--overrides",
-        json.dumps(pod),
-        "--image",
-        # Use ubuntu image so we get GNU rm and other tools
-        # Should match what we have in our pod definition
-        UBUNTU_IMAGE,
+        "exec",
+        "-it",
         pod_name,
         "--",
         "/bin/bash",
         "-l",
     ]
 
-    with cluster.auth():
-        subprocess.check_call(cmd)
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".json") as tmpf:
+        # Dump the pod spec to a temporary file
+        json.dump(pod, tmpf)
+        tmpf.flush()
+
+        with cluster.auth():
+            try:
+                # Ask kube-controller to create a pod
```

Comment:
```suggestion
                # Ask api-server to create a pod
```

Reply:
Can you also add a comment here on *why* we're turning this into two steps than 1?